src/tests/class_tests/openms/source/LogStream_test.cpp

```diff
@@ -72,6 +72,27 @@ class TestTarget
   bool notified;
 };
 
+// -----------------------------------
+//  Temporary measure to fix logging
+// -----------------------------------
+#define STRINGIFY(a) #a
+
+#define OPENMS_THREAD_CRITICAL(name) \
+    _Pragma( STRINGIFY( omp critical (name) ) )
+
+#define LOG_INFO \
+  OPENMS_THREAD_CRITICAL(oms_log) \
+  Log_info
```

Comment:
@dmccloskey I think you would have to change LogStream.h to include the `OPENMS_THREAD_CRITICAL(oms_log)` before every definition, can you try and do that? That should fix the issue

Reply:
I added the following to LogStream.h

```
#define STRINGIFY(a) #a

#define OPENMS_THREAD_CRITICAL(name) \
    _Pragma( STRINGIFY( omp critical (name) ) )

  /// Macro to be used if fatal error are reported (processing stops)
#define LOG_FATAL_ERROR \
  OPENMS_THREAD_CRITICAL(oms_log) \
  Log_fatal << __FILE__ << "(" << __LINE__ << "): "

  /// Macro to be used if non-fatal error are reported (processing continues)
#define LOG_ERROR \
  OPENMS_THREAD_CRITICAL(oms_log) \
  Log_error

  /// Macro if a warning, a piece of information which should be read by the user, should be logged
#define LOG_WARN \
  OPENMS_THREAD_CRITICAL(oms_log) \
  Log_warn

  /// Macro if a information, e.g. a status should be reported
#define LOG_INFO \
  OPENMS_THREAD_CRITICAL(oms_log) \
  Log_info

  /// Macro for general debugging information
#define LOG_DEBUG \
  OPENMS_THREAD_CRITICAL(oms_log) \
  Log_debug << __FILE__ << "(" << __LINE__ << "): "
```

However on windows the build chokes on `_Pragma`.  Where is `_Pragma` coming from?

Reply:
maybe you need to use `__pragma` on Windows? https://docs.microsoft.com/en-us/cpp/preprocessor/pragma-directives-and-the-pragma-keyword?view=vs-2017

Reply:
Thanks, that resulted in a successful build.  However, it does not look to fix the segmentation fault problem.  It still faults at line 89 `delete[] pbuf_;` in LogStream.cpp.